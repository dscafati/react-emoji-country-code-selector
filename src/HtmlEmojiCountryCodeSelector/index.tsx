import React, { SelectHTMLAttributes } from "react";
import data from "../data/dataset.csv";
import { Dataset } from "../types/dataset";

interface propsType extends SelectHTMLAttributes<HTMLSelectElement> {
  generateValue?(data: Dataset): string;
  generateLabel?(data: Dataset): string;
}

export default function HtmlEmojiCountrycodeSelector(props: propsType) {
  function getLabel(row: Dataset) {
    if (props.generateLabel) {
      return props.generateLabel(row);
    } else {
      return (
        <>
          {row.ISO} {row.flag} {row.code}
        </>
      );
    }
  }
  function getValue(row: Dataset) {
    if (props.generateValue) {
      return props.generateValue(row);
    } else {
      return row.code;
    }
  }
  return (
    <select {...props}>
      {props.children}
      {data.map((row: Dataset) => (
        <option key={row.ISO + row.code} value={getValue(row)}>
          {getLabel(row)}
        </option>
      ))}
    </select>
  );
}
