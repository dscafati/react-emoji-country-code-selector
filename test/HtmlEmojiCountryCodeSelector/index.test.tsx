import React from "react";
import ReactDOMServer from "react-dom/server";
import HtmlEmojiCountryCodeSelector from "../../src";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter() });

test("Snapshot test of the generated html code", () => {
  const component = (
    <HtmlEmojiCountryCodeSelector>
      <option>Default</option>
    </HtmlEmojiCountryCodeSelector>
  );
  const render = ReactDOMServer.renderToString(component);
  expect(render).toMatchSnapshot();
});

test("Events are called", () => {
  const mockCallback = jest.fn((e) => {});
  const component = shallow(
    <HtmlEmojiCountryCodeSelector onChange={mockCallback} />
  );
  component.simulate("change");
  expect(mockCallback).toBeCalledTimes(1);
});
