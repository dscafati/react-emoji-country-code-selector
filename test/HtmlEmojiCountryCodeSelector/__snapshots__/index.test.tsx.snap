// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of the generated html code 1`] = `"<select data-reactroot=\\"\\"><option>Default</option><option>undefined ğŸ‡¦ğŸ‡© undefined</option><option>undefined ğŸ‡¦ğŸ‡ª undefined</option><option>undefined ğŸ‡¦ğŸ‡« undefined</option><option>undefined ğŸ‡¦ğŸ‡¬ undefined</option><option>undefined ğŸ‡¦ğŸ‡® undefined</option><option>undefined ğŸ‡¦ğŸ‡± undefined</option><option>undefined ğŸ‡¦ğŸ‡² undefined</option><option>undefined ğŸ‡¦ğŸ‡´ undefined</option><option>undefined ğŸ‡¦ğŸ‡¶ undefined</option><option>undefined ğŸ‡¦ğŸ‡· undefined</option><option>undefined ğŸ‡¦ğŸ‡¸ undefined</option><option>undefined ğŸ‡¦ğŸ‡¹ undefined</option><option>undefined ğŸ‡¦ğŸ‡º undefined</option><option>undefined ğŸ‡¦ğŸ‡¼ undefined</option><option>undefined ğŸ‡¦ğŸ‡½ undefined</option><option>undefined ğŸ‡¦ğŸ‡¿ undefined</option><option>undefined ğŸ‡§ğŸ‡¦ undefined</option><option>undefined ğŸ‡§ğŸ‡§ undefined</option><option>undefined ğŸ‡§ğŸ‡© undefined</option><option>undefined ğŸ‡§ğŸ‡ª undefined</option><option>undefined ğŸ‡§ğŸ‡« undefined</option><option>undefined ğŸ‡§ğŸ‡¬ undefined</option><option>undefined ğŸ‡§ğŸ‡­ undefined</option><option>undefined ğŸ‡§ğŸ‡® undefined</option><option>undefined ğŸ‡§ğŸ‡¯ undefined</option><option>undefined ğŸ‡§ğŸ‡± undefined</option><option>undefined ğŸ‡§ğŸ‡² undefined</option><option>undefined ğŸ‡§ğŸ‡³ undefined</option><option>undefined ğŸ‡§ğŸ‡´ undefined</option><option>undefined ğŸ‡§ğŸ‡¶ undefined</option><option>undefined ğŸ‡§ğŸ‡· undefined</option><option>undefined ğŸ‡§ğŸ‡¸ undefined</option><option>undefined ğŸ‡§ğŸ‡¹ undefined</option><option>undefined ğŸ‡§ğŸ‡» undefined</option><option>undefined ğŸ‡§ğŸ‡¼ undefined</option><option>undefined ğŸ‡§ğŸ‡¾ undefined</option><option>undefined ğŸ‡§ğŸ‡¿ undefined</option><option>undefined ğŸ‡¨ğŸ‡¦ undefined</option><option>undefined ğŸ‡¨ğŸ‡¨ undefined</option><option>undefined ğŸ‡¨ğŸ‡© undefined</option><option>undefined ğŸ‡¨ğŸ‡« undefined</option><option>undefined ğŸ‡¨ğŸ‡¬ undefined</option><option>undefined ğŸ‡¨ğŸ‡­ undefined</option><option>undefined ğŸ‡¨ğŸ‡® undefined</option><option>undefined ğŸ‡¨ğŸ‡° undefined</option><option>undefined ğŸ‡¨ğŸ‡± undefined</option><option>undefined ğŸ‡¨ğŸ‡² undefined</option><option>undefined ğŸ‡¨ğŸ‡³ undefined</option><option>undefined ğŸ‡¨ğŸ‡´ undefined</option><option>undefined ğŸ‡¨ğŸ‡· undefined</option><option>undefined ğŸ‡¨ğŸ‡º undefined</option><option>undefined ğŸ‡¨ğŸ‡» undefined</option><option>undefined ğŸ‡¨ğŸ‡¼ undefined</option><option>undefined ğŸ‡¨ğŸ‡½ undefined</option><option>undefined ğŸ‡¨ğŸ‡¾ undefined</option><option>undefined ğŸ‡¨ğŸ‡¿ undefined</option><option>undefined ğŸ‡©ğŸ‡ª undefined</option><option>undefined ğŸ‡©ğŸ‡¯ undefined</option><option>undefined ğŸ‡©ğŸ‡° undefined</option><option>undefined ğŸ‡©ğŸ‡² undefined</option><option>undefined ğŸ‡©ğŸ‡´ undefined</option><option>undefined ğŸ‡©ğŸ‡¿ undefined</option><option>undefined ğŸ‡ªğŸ‡¨ undefined</option><option>undefined ğŸ‡ªğŸ‡ª undefined</option><option>undefined ğŸ‡ªğŸ‡¬ undefined</option><option>undefined ğŸ‡ªğŸ‡­ undefined</option><option>undefined ğŸ‡ªğŸ‡· undefined</option><option>undefined ğŸ‡ªğŸ‡¸ undefined</option><option>undefined ğŸ‡ªğŸ‡¹ undefined</option><option>undefined ğŸ‡«ğŸ‡® undefined</option><option>undefined ğŸ‡«ğŸ‡¯ undefined</option><option>undefined ğŸ‡«ğŸ‡° undefined</option><option>undefined ğŸ‡«ğŸ‡² undefined</option><option>undefined ğŸ‡«ğŸ‡´ undefined</option><option>undefined ğŸ‡«ğŸ‡· undefined</option><option>undefined ğŸ‡¬ğŸ‡¦ undefined</option><option>undefined ğŸ‡¬ğŸ‡§ undefined</option><option>undefined ğŸ‡¬ğŸ‡© undefined</option><option>undefined ğŸ‡¬ğŸ‡ª undefined</option><option>undefined ğŸ‡¬ğŸ‡« undefined</option><option>undefined ğŸ‡¬ğŸ‡¬ undefined</option><option>undefined ğŸ‡¬ğŸ‡­ undefined</option><option>undefined ğŸ‡¬ğŸ‡® undefined</option><option>undefined ğŸ‡¬ğŸ‡± undefined</option><option>undefined ğŸ‡¬ğŸ‡² undefined</option><option>undefined ğŸ‡¬ğŸ‡³ undefined</option><option>undefined ğŸ‡¬ğŸ‡µ undefined</option><option>undefined ğŸ‡¬ğŸ‡¶ undefined</option><option>undefined ğŸ‡¬ğŸ‡· undefined</option><option>undefined ğŸ‡¬ğŸ‡¸ undefined</option><option>undefined ğŸ‡¬ğŸ‡¹ undefined</option><option>undefined ğŸ‡¬ğŸ‡º undefined</option><option>undefined ğŸ‡¬ğŸ‡¼ undefined</option><option>undefined ğŸ‡¬ğŸ‡¾ undefined</option><option>undefined ğŸ‡­ğŸ‡° undefined</option><option>undefined ğŸ‡­ğŸ‡² undefined</option><option>undefined ğŸ‡­ğŸ‡³ undefined</option><option>undefined ğŸ‡­ğŸ‡· undefined</option><option>undefined ğŸ‡­ğŸ‡¹ undefined</option><option>undefined ğŸ‡­ğŸ‡º undefined</option><option>undefined ğŸ‡®ğŸ‡© undefined</option><option>undefined ğŸ‡®ğŸ‡ª undefined</option><option>undefined ğŸ‡®ğŸ‡± undefined</option><option>undefined ğŸ‡®ğŸ‡² undefined</option><option>undefined ğŸ‡®ğŸ‡³ undefined</option><option>undefined ğŸ‡®ğŸ‡´ undefined</option><option>undefined ğŸ‡®ğŸ‡¶ undefined</option><option>undefined ğŸ‡®ğŸ‡· undefined</option><option>undefined ğŸ‡®ğŸ‡¸ undefined</option><option>undefined ğŸ‡®ğŸ‡¹ undefined</option><option>undefined ğŸ‡¯ğŸ‡ª undefined</option><option>undefined ğŸ‡¯ğŸ‡² undefined</option><option>undefined ğŸ‡¯ğŸ‡´ undefined</option><option>undefined ğŸ‡¯ğŸ‡µ undefined</option><option>undefined ğŸ‡°ğŸ‡ª undefined</option><option>undefined ğŸ‡°ğŸ‡¬ undefined</option><option>undefined ğŸ‡°ğŸ‡­ undefined</option><option>undefined ğŸ‡°ğŸ‡® undefined</option><option>undefined ğŸ‡°ğŸ‡² undefined</option><option>undefined ğŸ‡°ğŸ‡³ undefined</option><option>undefined ğŸ‡°ğŸ‡µ undefined</option><option>undefined ğŸ‡°ğŸ‡· undefined</option><option>undefined ğŸ‡°ğŸ‡¼ undefined</option><option>undefined ğŸ‡°ğŸ‡¾ undefined</option><option>undefined ğŸ‡°ğŸ‡¿ undefined</option><option>undefined ğŸ‡±ğŸ‡¦ undefined</option><option>undefined ğŸ‡±ğŸ‡§ undefined</option><option>undefined ğŸ‡±ğŸ‡¨ undefined</option><option>undefined ğŸ‡±ğŸ‡® undefined</option><option>undefined ğŸ‡±ğŸ‡° undefined</option><option>undefined ğŸ‡±ğŸ‡· undefined</option><option>undefined ğŸ‡±ğŸ‡¸ undefined</option><option>undefined ğŸ‡±ğŸ‡¹ undefined</option><option>undefined ğŸ‡±ğŸ‡º undefined</option><option>undefined ğŸ‡±ğŸ‡» undefined</option><option>undefined ğŸ‡±ğŸ‡¾ undefined</option><option>undefined ğŸ‡²ğŸ‡¦ undefined</option><option>undefined ğŸ‡²ğŸ‡¨ undefined</option><option>undefined ğŸ‡²ğŸ‡© undefined</option><option>undefined ğŸ‡²ğŸ‡ª undefined</option><option>undefined ğŸ‡²ğŸ‡« undefined</option><option>undefined ğŸ‡²ğŸ‡¬ undefined</option><option>undefined ğŸ‡²ğŸ‡­ undefined</option><option>undefined ğŸ‡²ğŸ‡° undefined</option><option>undefined ğŸ‡²ğŸ‡± undefined</option><option>undefined ğŸ‡²ğŸ‡² undefined</option><option>undefined ğŸ‡²ğŸ‡³ undefined</option><option>undefined ğŸ‡²ğŸ‡´ undefined</option><option>undefined ğŸ‡²ğŸ‡µ undefined</option><option>undefined ğŸ‡²ğŸ‡¶ undefined</option><option>undefined ğŸ‡²ğŸ‡· undefined</option><option>undefined ğŸ‡²ğŸ‡¸ undefined</option><option>undefined ğŸ‡²ğŸ‡¹ undefined</option><option>undefined ğŸ‡²ğŸ‡º undefined</option><option>undefined ğŸ‡²ğŸ‡» undefined</option><option>undefined ğŸ‡²ğŸ‡¼ undefined</option><option>undefined ğŸ‡²ğŸ‡½ undefined</option><option>undefined ğŸ‡²ğŸ‡¾ undefined</option><option>undefined ğŸ‡²ğŸ‡¿ undefined</option><option>undefined ğŸ‡³ğŸ‡¦ undefined</option><option>undefined ğŸ‡³ğŸ‡¨ undefined</option><option>undefined ğŸ‡³ğŸ‡ª undefined</option><option>undefined ğŸ‡³ğŸ‡« undefined</option><option>undefined ğŸ‡³ğŸ‡¬ undefined</option><option>undefined ğŸ‡³ğŸ‡® undefined</option><option>undefined ğŸ‡³ğŸ‡± undefined</option><option>undefined ğŸ‡³ğŸ‡´ undefined</option><option>undefined ğŸ‡³ğŸ‡µ undefined</option><option>undefined ğŸ‡³ğŸ‡· undefined</option><option>undefined ğŸ‡³ğŸ‡º undefined</option><option>undefined ğŸ‡³ğŸ‡¿ undefined</option><option>undefined ğŸ‡´ğŸ‡² undefined</option><option>undefined ğŸ‡µğŸ‡¦ undefined</option><option>undefined ğŸ‡µğŸ‡ª undefined</option><option>undefined ğŸ‡µğŸ‡« undefined</option><option>undefined ğŸ‡µğŸ‡¬ undefined</option><option>undefined ğŸ‡µğŸ‡­ undefined</option><option>undefined ğŸ‡µğŸ‡° undefined</option><option>undefined ğŸ‡µğŸ‡± undefined</option><option>undefined ğŸ‡µğŸ‡² undefined</option><option>undefined ğŸ‡µğŸ‡³ undefined</option><option>undefined ğŸ‡µğŸ‡· undefined</option><option>undefined ğŸ‡µğŸ‡¸ undefined</option><option>undefined ğŸ‡µğŸ‡¹ undefined</option><option>undefined ğŸ‡µğŸ‡¼ undefined</option><option>undefined ğŸ‡µğŸ‡¾ undefined</option><option>undefined ğŸ‡¶ğŸ‡¦ undefined</option><option>undefined ğŸ‡·ğŸ‡ª undefined</option><option>undefined ğŸ‡·ğŸ‡´ undefined</option><option>undefined ğŸ‡·ğŸ‡¸ undefined</option><option>undefined ğŸ‡·ğŸ‡º undefined</option><option>undefined ğŸ‡·ğŸ‡¼ undefined</option><option>undefined ğŸ‡¸ğŸ‡¦ undefined</option><option>undefined ğŸ‡¸ğŸ‡§ undefined</option><option>undefined ğŸ‡¸ğŸ‡¨ undefined</option><option>undefined ğŸ‡¸ğŸ‡© undefined</option><option>undefined ğŸ‡¸ğŸ‡ª undefined</option><option>undefined ğŸ‡¸ğŸ‡¬ undefined</option><option>undefined ğŸ‡¸ğŸ‡­ undefined</option><option>undefined ğŸ‡¸ğŸ‡® undefined</option><option>undefined ğŸ‡¸ğŸ‡¯ undefined</option><option>undefined ğŸ‡¸ğŸ‡° undefined</option><option>undefined ğŸ‡¸ğŸ‡± undefined</option><option>undefined ğŸ‡¸ğŸ‡² undefined</option><option>undefined ğŸ‡¸ğŸ‡³ undefined</option><option>undefined ğŸ‡¸ğŸ‡´ undefined</option><option>undefined ğŸ‡¸ğŸ‡· undefined</option><option>undefined ğŸ‡¸ğŸ‡¸ undefined</option><option>undefined ğŸ‡¸ğŸ‡¹ undefined</option><option>undefined ğŸ‡¸ğŸ‡» undefined</option><option>undefined ğŸ‡¸ğŸ‡½ undefined</option><option>undefined ğŸ‡¸ğŸ‡¾ undefined</option><option>undefined ğŸ‡¸ğŸ‡¿ undefined</option><option>undefined ğŸ‡¹ğŸ‡¨ undefined</option><option>undefined ğŸ‡¹ğŸ‡© undefined</option><option>undefined ğŸ‡¹ğŸ‡« undefined</option><option>undefined ğŸ‡¹ğŸ‡¬ undefined</option><option>undefined ğŸ‡¹ğŸ‡­ undefined</option><option>undefined ğŸ‡¹ğŸ‡¯ undefined</option><option>undefined ğŸ‡¹ğŸ‡° undefined</option><option>undefined ğŸ‡¹ğŸ‡± undefined</option><option>undefined ğŸ‡¹ğŸ‡² undefined</option><option>undefined ğŸ‡¹ğŸ‡³ undefined</option><option>undefined ğŸ‡¹ğŸ‡´ undefined</option><option>undefined ğŸ‡¹ğŸ‡· undefined</option><option>undefined ğŸ‡¹ğŸ‡¹ undefined</option><option>undefined ğŸ‡¹ğŸ‡» undefined</option><option>undefined ğŸ‡¹ğŸ‡¼ undefined</option><option>undefined ğŸ‡¹ğŸ‡¿ undefined</option><option>undefined ğŸ‡ºğŸ‡¦ undefined</option><option>undefined ğŸ‡ºğŸ‡¬ undefined</option><option>undefined ğŸ‡ºğŸ‡² undefined</option><option>undefined ğŸ‡ºğŸ‡¸ undefined</option><option>undefined ğŸ‡ºğŸ‡¾ undefined</option><option>undefined ğŸ‡ºğŸ‡¿ undefined</option><option>undefined ğŸ‡»ğŸ‡¦ undefined</option><option>undefined ğŸ‡»ğŸ‡¨ undefined</option><option>undefined ğŸ‡»ğŸ‡ª undefined</option><option>undefined ğŸ‡»ğŸ‡¬ undefined</option><option>undefined ğŸ‡»ğŸ‡® undefined</option><option>undefined ğŸ‡»ğŸ‡³ undefined</option><option>undefined ğŸ‡»ğŸ‡º undefined</option><option>undefined ğŸ‡¼ğŸ‡« undefined</option><option>undefined ğŸ‡¼ğŸ‡¸ undefined</option><option>undefined ğŸ‡¾ğŸ‡ª undefined</option><option>undefined ğŸ‡¾ğŸ‡¹ undefined</option><option>undefined ğŸ‡¿ğŸ‡¦ undefined</option><option>undefined ğŸ‡¿ğŸ‡² undefined</option><option>undefined ğŸ‡¿ğŸ‡¼ undefined</option></select>"`;
